### PR TITLE
Fix reference to return produced by ApplicativeDo

### LINKED ISSUE #6820: ApplicativeDo produces interface error in some cases
Minimal example
```
{-# LANGUAGE ApplicativeDo #-}
module Broken where

import Daml.Script

template T
  with
    p : Party
  where
    signatory p
    choice C : ()
      controller p
      do pure ()

broken : Script ()
broken = do
  p <- allocateParty "p"
  -- this redundant create matters for the error
  _ <- submit p (createCmd (T p))
  cid <- submit p (createCmd (T p))
  submit p (exerciseCmd cid C)
  -- this let matters as well
  let _ = ()
  return ()
```

Error
```
  daml/Broken.daml:21:3: error:
  • Can't find interface-file declaration for variable GHC.Base.return
  Probable cause: bug in .hi-boot file, or inconsistent .hi file
  Use -ddump-if-trace to get an idea of which file caused the error
  • In the expression:
  do p <- allocateParty "p"
  _ <- submit p (createCmd (T p))
  cid <- submit p (createCmd (T p))
  submit p (exerciseCmd cid C)
  let _ = ()
  return ()
  In an equation for ‘broken’:
  broken
  = do p <- allocateParty "p"
  _ <- submit p (createCmd (T p))
  cid <- submit p (createCmd (T p))
  submit p (exerciseCmd cid C)
  let _ = ...
  return ()
```

Removing `ApplicativeDo` (which is not needed here) fixes the issue.

### PR DESCRIPTION
**Pull request by cocreature**
see https://github.com/digital-asset/ghc/pull/53 for details.

fixes #6820

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ci/da-ghc-lib/compile.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
   variables:
     ghc-lib-sha: '77fcc42eef2256ba9bb61a7e442a65bedb19a35d'
     base-sha: '9c787d4d24f2b515934c8503ee2bbd7cfac4da20'
-    patches: 'f844cb345412f6a9fd5708d8b697fd0800dd0909 833ca63be2ab14871874ccb6974921e8952802e9'
+    patches: '80f2ae67ec69d79c116b1e239228635dc460e783 833ca63be2ab14871874ccb6974921e8952802e9'
```
Comment (cocreature) [inline ci/da-ghc-lib/compile.yml:15]: I’ll bump this again once the PR to our GHC fork is merged but I wanted to see if CI passes.

**Comment by garyverhaegen-da** [comment]
Cross reference to save future readers a bit of digging: https://github.com/digital-asset/ghc/pull/53

**Comment by cocreature** [comment]
@garyverhaegen-da linked in the PR description already :slightly_smiling_face:

**Comment by garyverhaegen-da** [comment]
> @garyverhaegen-da linked in the PR description already 🙂

Ah, right. I don't know how I missed that 😓

File: compiler/damlc/tests/daml-test-files/ApplicativeDo.daml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+-- @INFO Use let
+-- @INFO Redundant pure
+module ApplicativeDo where
+
+-- This used to produce a reference to GHC.Base.return
+-- which resulted in an interface error, see
+-- https://github.com/digital-asset/daml/issues/6820
+referenceReturn : Update ()
+referenceReturn = do
+  -- this redundant create matters for the error
```
Comment (hurryabit) [inline compiler/damlc/tests/daml-test-files/ApplicativeDo.daml:11]:
```suggestion
  -- this redundant pure matters for the error
```

**Pull request merged**